### PR TITLE
bounty submission and admin bounty management dashboard

### PR DESCRIPTION
**Pull request by nischal-shetty2**
reopen #1091 after adding auth as requested here https://github.com/code100x/cms/pull/1091#issuecomment-2347099435

also made it more responsive  

https://github.com/user-attachments/assets/c19cc897-20ce-4952-8055-fb91c70b8436

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I assure there is no similar/duplicate pull request regarding same issue

### CODE REVIEW

**Comment by nischal-shetty2** [comment]
@devsargam @hkirat reopened it here. need to be reviewed!

**Comment by nischal-shetty2** [comment]
@devsargam @hkirat @siinghd any updates on this? Needs to be reviewed

**Comment by devsargam** [comment]
includes db changes will do later

**Comment by nischal-shetty2** [comment]
> includes db changes will do later

reminder @devsargam ser

**Comment by nischal-shetty2** [comment]
hey @devsargam @hkirat its been a minute🫣

**Comment by hkirat** [comment]
This is a good one we should has out @devsargam 
Dying to give out bounties in cohort 3

File: src/app/Bounty/page.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import { SolanaAddress, UpiId, BountySubmission } from '@prisma/client';
+import BountySubmissionDialog from '@/components/bounty/BountySubmissionDialog';
+import { getPayoutMethods } from '@/actions/payoutMethods';
+import { getUserBounties } from '@/actions/bounty';
+import { toast } from 'sonner';
+import { Skeleton } from '@/components/ui/skeleton';
+
+export default function Page() {
+  const [isBountyDialogOpen, setIsBountyDialogOpen] = useState<boolean>(false);
+  const [upiAddresses, setUpiAddresses] = useState<UpiId[] | undefined>([]);
```
Comment (devsargam) [inline src/app/Bounty/page.tsx:14]: Small nitpick here why use | undefined? 🤔

File: src/app/Bounty/page.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  const [solanaAddresses, setSolanaAddresses] = useState<
+    SolanaAddress[] | undefined
```
Comment (devsargam) [inline src/app/Bounty/page.tsx:16]: same as above

File: src/components/PaymentDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+  const handleUpiDelete = (id: number) => {
+    executeDeleteUPI({ id });
+  };
+
+  const handleSolanaDelete = (id: number) => {
+    executeDeleteSolana({ id });
+  };
+
+  useEffect(() => {
```
Comment (devsargam) [inline src/components/PaymentDropdown.tsx:54]: This effect seems unnecessary.

**Review by devsargam — COMMENTED** [review]
LGTM!!!

**Comment by devsargam** [comment]
@nischal-shetty2 this took too long to review. Sorry for that I was too busy/lazy last couple of weeks. 

Honestly speaking loved this pr and your code quality is over the top ✨

**Comment by devsargam** [comment]
Just make the changes suggested above and I think the pr is ready for review

File: src/app/Bounty/page.tsx (hunk 2 above)
Comment (nischal-shetty2) [reply src/app/Bounty/page.tsx:16]: Thought process what that if user hasn't entered the address then it stays undefined, but youre right we check if address exists anyway so theres no point in this. ill remove them both.

File: src/components/PaymentDropdown.tsx (hunk 3 above)
Comment (nischal-shetty2) [reply src/components/PaymentDropdown.tsx:54]: first i had the payment dialog open by default so this was required, now the only relevant updates to payout methods happen as a result of dialog interactions so this is irrelevant. tested it by removing this useeffect will push the changes now

**Comment by nischal-shetty2** [comment]
@devsargam fixed them, safe to merge if no other changes required

**Review by devsargam — APPROVED** [review]
LTGM

**Comment by devsargam** [comment]
Thank you sir. @hkirat might want to test out bounties starting here :p

**Comment by nischal-shetty2** [comment]
> Thank you sir. @hkirat might want to test out bounties starting here :p

it does need to be tested for someone first, might as well be me @hkirat 😂

**Pull request merged**